### ISSUE TITLE
Should not report duplicate declaration errors for locals and parameters with the same name

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#8834]
```ts
class C {
  constructor(options?: number) {
    var options = (options || 0);
    // !!! error TS2687: All declarations of 'options' must have identical modifiers.
  }
}
```
